mllib/src/main/scala/org/apache/spark/ml/feature/Imputer.scala

```diff
@@ -75,6 +75,8 @@ private[feature] trait ImputerParams extends Params with HasInputCols {
 
   /** Validates and transforms the input schema. */
   protected def validateAndTransformSchema(schema: StructType): StructType = {
+    require(get(inputCols).isDefined, "Input cols must be defined first.")
```

Comment:
As I mentioned in #17316, is this really required? Since a non-set param for these will in any case throw an exception during `transformSchema` (or `fit`, or `transform`) with "no default value found"